postgis/gserialized_gist_nd.c

```diff
@@ -180,15 +179,16 @@ gidx_merge(GIDX **b_union, GIDX *b_new)
 
 	POSTGIS_DEBUGF(4, "merging gidx (%s) into gidx (%s)", gidx_to_string(b_new), gidx_to_string(*b_union));
 
-	if (dims_new > dims_union)
+	/* Shrink unshared dimensions */
+	if (dims_new < dims_union)
```

Comment:
Is this shrinking really necessary now? If b_union has 10 dimensions and b_new 2, I'd expect the merge to have 10, not 2; that is, a merge between 2 boxes should be bigger, never smaller.

Reply:
Think of 3DM box: it has (minx, maxx, miny, maxy, -FLT_MAX, FLT_MAX, minm, maxm ) for padding, it is a 4D gidx.

When this gets into calculations here, it resets whatever was there in Z to -flt_max / flt_max, in gidx_merge logic. That is equivalent of dropping the dimension. Thus, unshared dimensions should be dropped to correctly summarize a group of other mixed-dimensions boxes. 

Is there a fault in this logic?

Reply:
> Think of 3DM box: it has (minx, maxx, miny, maxy, -FLT_MAX, FLT_MAX, minm, maxm ) for padding, it is a 4D gidx.

That's what I wasn't expecting. I'd expect that a XYM GBOX would result in a 3-dimension gidx, but as you say it's done by adding an infinite dimension instead [1]. Based on the preferences taken ([2]), I'm unsure whether `gidx_merge` the union should contain that extra dimension or not, but it appears that this was in fact the issue here.

[1] https://github.com/postgis/postgis/blob/svn-trunk/libpgcommon/gserialized_gist.c#L275
[2] https://lists.osgeo.org/pipermail/postgis-devel/2015-February/024759.html